src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6675,38 +6739,174 @@ register_handler(int sig, struct sigaction *old_handler)
+	if (buf == NULL) {
+		fprintf(stderr, "Fail to allocate memory for buf %d (%s)\n", errno,
+			strerror(errno));
+		exit(1);
+	}
+	readsize = fread(buf, 1, CMDLINE_BUF_SIZE, f);
+	if (readsize <= 0) {
+		fprintf(stderr, "Fail to read /proc/self/cmdline %d (%s)\n", errno,
+			strerror(errno));
+		fclose(f);
+		exit(1);
+	}
+
+	fclose(f);
 
-	readsize = readlink("/proc/self/exe", exe_path, DFS_MAX_PATH);
-	if (readsize >= DFS_MAX_PATH) {
-		DS_ERROR(ENAMETOOLONG, "path from readlink() is too long");
-		goto out;
-	} else if (readsize < 0) {
-		DS_ERROR(errno, "readlink() failed");
-		goto out;
+	exe_short_name = basename(buf);
+	if (exe_short_name == NULL) {
+		fprintf(stderr, "Fail to determine exe_short_name %d (%s)\n", errno,
+			strerror(errno));
+		exit(1);
+	}
+	exe_short_name = strndup(exe_short_name, DFS_MAX_NAME);
+	if (exe_short_name == NULL) {
+		printf("Fail to allocate exe_short_name %d (%s)\n", errno, strerror(errno));
+		exit(1);
+	}
+	first_arg = NULL;
+	end       = buf + readsize;
+
+	for (p = buf; p < end;) {
+		if (count == 1) {
+			if (p[0] == '/' || memcmp(p, "./", 2) == 0 || memcmp(p, "../", 3) == 0) {
+				/* Extract the first argument in command line */
+				first_arg = basename(p);
+				if (first_arg == NULL) {
+					fprintf(stderr, "Fail to determine first_arg %d (%s)\n",
+						errno, strerror(errno));
+					exit(1);
+				}
+				first_arg = strndup(first_arg, DFS_MAX_NAME);
+				if (first_arg == NULL) {
+					fprintf(stderr, "Fail to allocate first_arg %d (%s)\n",
+						errno, strerror(errno));
+					exit(1);
+				}
+			}
+			break;
+		}
+		count++;
+		while (*p++)
+			;
 	}
+	free(buf);
+	/* We allocated buffers for exe_short_name and first_arg. Now buf can be deallocated. */
+}
+
+static char *bypass_bash_cmd_list[] = {"autoconf", "configure", "libtool", "libtoolize",
+				       "lsb_release"};
 
-	exe_short = basename(exe_path);
-	if (exe_short == NULL) {
-		DS_ERROR(errno, "basename() failed");
-		goto out;
+static char *bypass_python3_cmd_list[] = {"scons", "scons-3", "dnf", "dnf-3", "meson"};
+
+static char *bypass_app_list[] = {"arch", "as", "awk", "basename", "bc", "cal", "cat", "chmod",
+				      "chown", "clang", "clear", "cmake", "cmake3", "cp", "cpp",
+				      "daos", "daos_agent", "daos_engine", "daos_server", "df",
+				      "dfuse", "dmg", "expr", "f77", "f90", "f95", "file", "gawk",
+				      "gcc", "gfortran", "gmake", "go", "gofmt", "grep", "g++",
+				      "head", "link", "ln", "ls", "kill", "m4", "make", "mkdir",
+				      "mktemp", "mv", "nasm", "yasm", "nm",  "numactl", "patchelf",
+				      "ping", "pkg-config", "ps", "pwd", "ranlib", "readelf",
+				      "readlink", "rename", "rm", "rmdir", "rpm", "sed", "seq",
+				      "size", "sleep", "sort", "ssh", "stat", "strace", "strip",
+				      "su", "sudo", "tail", "tee", "telnet", "time", "top", "touch",
+				      "tr", "truncate", "uname", "vi", "vim", "whoami", "yes"};
+
+static void
+check_whitelist(void)
```

Comment:
can you please confirm that interception is turned off for everything in that list and user's list for both parent and child processes?

Reply:
Yes. I did confirm it.
I can add debug log if D_IL_REPORT is true and check the log in ftest.

Reply:
Interception status (ON/OFF) is checked in the test inside src/tests/ftest/dfuse/il_whitelist.py. "ls" is in built-in bypass list and "whitelist_test" is in user supplied bypass list.